drivers/virtualbox/virtualbox_test.go

```diff
@@ -1,10 +1,207 @@
+}
+
+type VBoxManagerMock struct {
+	expectedArgs string
+	output       string
+	stdErr       string
+	err          error
+}
+
+func (v *VBoxManagerMock) queryVirtualBox(args ...string) (string, string, error) {
+	if strings.Join(args, " ") == v.expectedArgs {
+		return v.output, v.stdErr, v.err
+	}
+	return "", "", errors.New("Invalid args")
+}
+
+func TestStateRunning(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="running"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Running, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStatePaused(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="paused"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Paused, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateSaved(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="saved"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Saved, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateStopped(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="poweroff"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Stopped, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateAborted(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="aborted"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Stopped, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateUnknown(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="whatever"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.None, machineState)
+	assert.NoError(t, err) // Really?
+}
+
+func TestStateInvalid(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState=`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.None, machineState)
+	assert.NoError(t, err) // Really?
+}
+
+func TestStateOfMachineNotFound(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		stdErr:       "Could not find a registered machine named 'unknown'",
+		err:          errors.New("Bug"),
```

Comment:
As I said, I'm always reluctant in sharing consts between tests and code. It make the test less useful. Or maybe I didn't understand what you said.